From 0364a64b25fe7d8f6e632fd548b57b508aed6416 Mon Sep 17 00:00:00 2001
From: Adrian Kalaszi <akalaszi@cloudera.com>
Date: Mon, 22 Feb 2016 02:23:56 +0100
Subject: [PATCH 130/140] KITE-1114: fix test

---
 .../TestTemporaryFileSystemDatasetRepository.java  |   46 +++++---------------
 1 file changed, 12 insertions(+), 34 deletions(-)

diff --git a/kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestTemporaryFileSystemDatasetRepository.java b/kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestTemporaryFileSystemDatasetRepository.java
index 3df4757..12d4e53 100644
--- a/kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestTemporaryFileSystemDatasetRepository.java
+++ b/kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestTemporaryFileSystemDatasetRepository.java
@@ -16,46 +16,24 @@
 
 package org.kitesdk.data.spi.filesystem;
 
-import java.io.IOException;
-import java.net.URI;
-
-import org.apache.avro.generic.GenericData.Record;
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.fs.Path;
-import org.junit.Assert;
-import org.junit.Test;
-import org.kitesdk.data.Dataset;
 import org.kitesdk.data.TestDatasetRepositories;
 import org.kitesdk.data.spi.DatasetRepository;
 import org.kitesdk.data.spi.MetadataProvider;
 
-public class TestTemporaryFileSystemDatasetRepository
-    extends TestDatasetRepositories {
-
-  public TestTemporaryFileSystemDatasetRepository(boolean distributed) {
-    super(distributed);
-  }
-
-  public static String KEY = "key";
-
-  @Test
-  public void verifyIfRequiredTmpFileSystemDatasetRepositoryHasBeenCreated()
-      throws IOException {
+public class TestTemporaryFileSystemDatasetRepository extends TestDatasetRepositories {
 
-    Dataset<Record> created = repo.create(NAMESPACE, NAME, testDescriptor);
-    URI location = created.getDescriptor().getLocation();
-    System.err.println(location);
-    Assert.assertNotNull(
-        "FileSystemDatasetRepository should return descriptor locations",
-        location);
-    Assert.assertTrue("Dataset data directory:" + location + " should exist",
-        fileSystem.exists(new Path(location)));
+	public TestTemporaryFileSystemDatasetRepository(boolean distributed) {
+		super(distributed);
+	}
 
-  }
+	public static String KEY = "key";
 
-  @Override
-  public DatasetRepository newRepo(MetadataProvider provider) {
-    return new TemporaryFileSystemDatasetRepository(new Configuration(),
-        testDirectory, NAMESPACE, KEY);
-  }
+	@Override
+	public DatasetRepository newRepo(MetadataProvider provider) {
+		TemporaryFileSystemDatasetRepository sut = new TemporaryFileSystemDatasetRepository(new Configuration(),
+				testDirectory, NAMESPACE, KEY);
+		this.testProvider = sut.getMetadataProvider();
+		return sut;
+	}
 }
-- 
1.7.9.5

